### PR TITLE
feat : feature related to post added

### PR DESCRIPTION
**Pull request by parulgupta2001**
1.Made single post page to see the details of single post.
2.shown all the posts on home page.
3.User can add the post.
4.user can add the comment on any post.

### CODE REVIEW

File: src/frontend/components/comment-modal/Comment-modal.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  const addCommentHandler = () => {
+    dispatch(addComment({ postId, commentData: { text }, token }));
+    setCommentModal(false);
+  };
+
+  return (
+    <div className="comment_modal_container">
+      <img src={singlePost.avatarURL} className="avatar_img" />
```
Comment (sri-rishi) [inline src/frontend/components/comment-modal/Comment-modal.js:31]: Suggestion: provide `alt` attribute to img tag, helps in accessibility

File: src/frontend/components/left-navbar/Left-navbar.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import "./left-navbar.css";
+import { FaFeatherAlt } from "react-icons/fa";
+import { AiFillHome } from "react-icons/ai";
+import { RiHashtag } from "react-icons/ri";
+import { FiBookmark } from "react-icons/fi";
+import { CgProfile } from "react-icons/cg";
```
Comment (sri-rishi) [inline src/frontend/components/left-navbar/Left-navbar.js:6]: Suggestion: you can import all icons in one and export from there

File: src/frontend/components/left-navbar/Left-navbar.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import { Link } from "react-router-dom";
+import { PostModal } from "../index";
+import { useDispatch, useSelector } from "react-redux";
+import { addPost } from "../../features/post-slice";
+
+export function LeftNav({ postModal, setPostModal }) {
+  const token = useSelector((store) => store.auth.token);
```
Comment (sri-rishi) [inline src/frontend/components/left-navbar/Left-navbar.js:13]: Suggestion: It's my opinion it's good to use optional chaining (?) to check that each reference in the chain is valid, you can differ

File: src/frontend/features/auth-slice.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    [signup.fulfilled]: (state, action) => {
+      state.token = action.payload;
+      state.status = "fulfilled";
+      localStorage.setItem("token", action.payload);
+    },
+    [signup.rejected]: (state, action) => {
+      state.status = "error";
+      state.error = action.error;
+      console.log(state.error);
```
Comment (sri-rishi) [inline src/frontend/features/auth-slice.js:48]: Suggestion: for error you can use `console.error`

**Review by sri-rishi — APPROVED** [review]
Code looks  good to merge, one suggestion provide preview link also

File: src/backend/db/users.js [hunk 5]
```diff
@@ -5,14 +5,353 @@ import { formatDate } from "../utils/authUtils";
...
+    userWebsite: "https://www.bengalurufc.com/player/sunil-chhetri/",
+  },
+
+  //   {
+  //     _id: uuid(),
+  //     firstName: "Garry",
+  //     lastName: "Tan",
+  //     username: "garry",
+  //     password: "garrytan",
+  //     createdAt: formatDate(),
```
Comment (simran61) [inline src/backend/db/users.js:279]: Can remove extra comments if not in use.

**Review by simran61 — APPROVED** [review]
Looks good

File: src/App.js [hunk 6]
```diff
@@ -1,41 +1,26 @@
 import "./App.css";
-import logo from "./logo.png";
+import { Routes, Route } from "react-router-dom";
+import { Home, Login, Signup, SinglePost } from "./frontend/pages/index";
+import { RequiresAuth } from "./frontend/requires-auth";
```
Comment (HeyNitin) [inline src/App.js:4]: You can have a look at absolute imports for next project

File: src/App.js [hunk 7]
```diff
@@ -1,41 +1,26 @@
...
-        </div>
-      </header>
+    <div className="app">
+      <Routes>
+        <Route
+          path="/"
+          element={
+            // <RequiresAuth>
+            <Home />
+            // </RequiresAuth>
```
Comment (HeyNitin) [inline src/App.js:16]: Remove commented code if not necessary

File: src/backend/db/users.js (hunk 5 above)
Comment (HeyNitin) [inline src/backend/db/users.js:279]: remove commented code

File: src/frontend/components/comment-modal/Comment-modal.js (hunk 1 above)
Comment (HeyNitin) [inline src/frontend/components/comment-modal/Comment-modal.js:31]: as suggested by other users as well, use alt in image tag

File: src/frontend/components/comment-modal/Comment-modal.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+      <div>
+        <input
+          onChange={changeHandler}
+          value={text}
+          placeholder="Post your reply"
+        />
+
```
Comment (HeyNitin) [inline src/frontend/components/comment-modal/Comment-modal.js:39]: remove extra spaces

**Review by HeyNitin — APPROVED** [review]
Remove commented code if not necessary and extra spaces too.
Left some suggestions, please have a look, rest looks good

**Pull request merged**